### ISSUE TITLE
[Offline] If a push has errors, do a pull and show dialog

### ISSUE THREAD

**Issue by knipec** [techmatters/terraso-mobile-client#2293]
### Acceptance Criteria
- If a push fully succeeds, no need to pull right after.
- If a push response has errors (like an entity was deleted or permissions were restricted), pull data from server. This will cause the user's incompatible data to be dropped and clobbered with server data. But at least the data in the app will be in a consistent state. And if the push attempt stays in a history log we think that's ok. We suspect this case would happen infrequently anyway.
- If this happens when app starts or when user comes back online, we will already always pull after a push (from [#2292](https://github.com/techmatters/terraso-mobile-client/issues/2292)), so no need to do it an extra time if the push has errors
- Do this for soil data pushes through the new mechanism

Also:
- Notify the user of data conflicts per #2324, or create a new task for this:
If the user client pushes data and
  - A site was deleted or removed from a project
  - The user was removed from a project
  - Depth requirements for a project changed
....then display the dialog
- Consider if any mutation's error response should trigger a pull, including things that don't go through the new mechanism (like site notes, project edits, etc). If not, what are the consequences? And do we care if this case shows an error message?

**Comment by tm-ruxandra** [comment]
Blocked by #2324

**Comment by CourtneyLee333** [comment]
I'm assuming this will be in v217.

**Comment by DerekCaelin** [comment]
Courtney and I tested. This passes our tests! We:
- Deleted a site while data collector was offline
- Transferred a site while data collector was offline
- Changed user's role they were offline

It all behaved as expected.
